examples/next/src/lib/types/providers.ts

```diff
@@ -0,0 +1,22 @@
+import { MemoryClient } from 'tevm';
+import { Chain as BaseChain, PublicClient } from 'viem';
+
+/**
+ * @type {Chain}
+ * @notice A Viem chain with a custom rpc url and a public client
+ * @dev The rpc url will be used to create a Tevm memory client from a fork.
+ * @dev The provider is useful to fetch ABIs with WhatsABI.
+ */
+export type Chain = BaseChain & {
+  custom: {
+    rpcUrl: string;
+    provider: PublicClient;
+  };
+};
+
+/**
+ * @type {Client}
+ * @notice A Tevm memory client
+ * @dev This will be used to write/read to/from the chain's state.
+ */
+export type Client = MemoryClient;
```

Comment:
Not really, I just like to keep such types in one file, so it can be updated in one place if we want to change to another client later, instead of skimming through all files.

But maybe we do want to skim through all the files actually. I'll update!

Reply:
don't bother was just confused